src/endpoint/s3/ops/s3_get_object_attributes.js

```diff
@@ -0,0 +1,124 @@
+    const key_b64 = req.headers[`x-amz-server-side-encryption-key`];
+    const key_md5_b64 = req.headers[`x-amz-server-side-encryption-key-md5`];
+    if (algorithm || key_b64 || key_md5_b64) {
+        dbg.error('get_object_attributes: The x-amz-server-side-encryption header is used on' +
+            'PUT object and it is not valid for GET request');
+        throw new S3Error(S3Error.BadRequest);
+    }
+    const encryption_customer = s3_utils.parse_sse_c(req);
+    return encryption_customer;
+}
+
+/**
+ * _parse_attributes parses the header in which the attributes are passed as tring with ',' as separator
+ * and returns array with the the attributes according to the valid attributes list (otherwise it throws an error)
+ * @param {nb.S3Request} req
+ * @returns {string[]}
+ */
+function _parse_attributes(req) {
+    const attributes_str = req.headers['x-amz-object-attributes'];
+    if (!attributes_str) {
+        dbg.error('get_object_attributes: must pass at least one attribute from:',
+            s3_utils.OBJECT_ATTRIBUTES);
+        throw new S3Error(S3Error.InvalidArgument);
+    }
+    const attributes = attributes_str.split(',');
+    const all_valid = attributes.every(item => s3_utils.OBJECT_ATTRIBUTES.includes(item));
+    if (!all_valid) {
+        dbg.error('get_object_attributes: received attributes:', attributes,
+            'at least one of the attributes is not from:', s3_utils.OBJECT_ATTRIBUTES);
+        throw new S3Error(S3Error.InvalidArgument);
+    }
+    return attributes;
+}
+
+function _set_response_headers(req, res, reply, version_id) {
+    if (version_id) {
+        res.setHeader('x-amz-version-id', version_id);
+        if (reply.delete_marker) {
+            res.setHeader('x-amz-delete-marker', 'true');
+        }
+    }
+    if (reply.last_modified_time) {
+        res.setHeader('Last-Modified', time_utils.format_http_header_date(new Date(reply.last_modified_time)));
+    } else {
+        res.setHeader('Last-Modified', time_utils.format_http_header_date(new Date(reply.create_time)));
+    }
+    s3_utils.set_encryption_response_headers(req, res, reply.encryption);
+}
+
+/**
+ * _parse_reply_according_to_attributes currently the reply is md_object
+ * and we return the properties according to the attributes the client asked for
+ * @param {object} reply
+ * @param {object} attributes
+ * @returns {object}
+ */
+function _parse_reply_according_to_attributes(reply, attributes) {
+    const reply_without_filter = {
+        ETag: `"${reply.etag}"`,
+        // Checksum: '', // GAP
+        // ObjectParts: '', // GAP
+        StorageClass: reply.storage_class,
+        ObjectSize: reply.size
+    };
+    const filtered_reply = {
+        GetObjectAttributesOutput: {
+        }
+    };
+    for (const key of attributes) {
+        if (key in reply_without_filter) {
+            filtered_reply.GetObjectAttributesOutput[key] = reply_without_filter[key];
+        }
```

Comment:
else print warning that the reply is not including all the requested attributes.

Reply:
I added the printing, but it might also be printed in namespace s3, where the server itself did add them.
I tested (in AWS): `aws s3api get-object-attributes --bucket shira-v --key cat.txt --version-id a92uCOZJSGaTCp2E8GARjXR02zLANcf5  --object-attributes "StorageClass" "ETag" "ObjectSize" "Checksum" "ObjectParts"`
The output:
```
{
    "LastModified": "2024-10-01T08:39:27+00:00",
    "VersionId": "a92uCOZJSGaTCp2E8GARjXR02zLANcf5",
    "ETag": "d41d8cd98f00b204e9800998ecf8427e",
    "StorageClass": "STANDARD",
    "ObjectSize": 0
}
```
You can see there is no `Checksum` and `ObjectParts` in the output.